utils.py

```diff
@@ -759,96 +759,210 @@ def melt_kt_grouped(kt, columns_to_melt, value_column_names, key_column_name='va
+                   k2_col: str = 'k2',
+                   sex_s_col: str = 's',
+                   sex_sex_col: str = 'sex',
+                   first_degree_threshold: Tuple[float, float] = (0.23, 0.27),  # TODO: Look at distribution
+                   second_degree_threshold: Tuple[float, float] = (0.12, 0.13),  # TODO: Look at distribution
+                   k2_parent_offspring_threshold: float = 0.1  # TODO: Look at distribution
+                   ) -> hl.Pedigree:
+    """
+
+    Infers familial relationships from the results of pc_relate and sex information.
+    Note that both kinship and k2 are needed in the pc_relate output.
+
+    This function returns a pedigree containing trios inferred from the data. Family ID can be the same for multiple
+    trios if one or more members of the trios are related (e.g. sibs, multi-generational family).
+
+    Note that this function only returns complete trios defined as:
+    one child, one father and one mother (sex is required for both parents)
+
+    :param Table kin_ht: pc_relate output table
+    :param Table sex_ht: A table containing sex for each sample
+    :param set of str duplicated_samples: Duplicated samples to remove (If not provided, this function won't work as it assumes that each child has exactly two parents)
+    :param str i_col: Column containing the 1st sample id in the pc_relate table
+    :param str j_col: Column containing the 1st sample id in the pc_relate table
+    :param str kin_col: Column containing the kinship in the pc_relate table
+    :param str k2_col: Column containing kin2 in the pc_relate table
+    :param str sex_s_col: Column containing the sample id in the sex table
+    :param str sex_sex_col: Column containing the sample sex in the sex table
+    :param (float, float) first_degree_threshold: Lower/upper bounds for kin for 1st degree relatives
+    :param (float, float) second_degree_threshold: Lower/upper bounds for kin for 2nd degree relatives
+    :param float k2_parent_offspring_threshold: Upper bound on kin2 for a parent/offspring
+    :return: Pedigree containing all trios in the data
+    :rtype: Pedigree
+    """
+
+    def get_fam_samples(sample: str,
+                        fam: Set[str],
+                        samples_rel: Dict[str, Set[str]],
+                        ) -> Set[str]:
+        """
+        Given a sample, its known family and a dict that links samples with their relatives, outputs the set of
+        samples that constitute this sample family.
+
+        :param str sample: sample
+        :param dict of str -> set of str samples_rel: dict(sample -> set(sample_relatives))
+        :param set of str fam: sample known family
+        :return: Family including the sample
+        :rtype: set of str
+        """
+        fam.add(sample)
+        for s2 in samples_rel[sample]:
             if s2 not in fam:
-                get_fam_samples(s2, fam, samples_rel)
+                return get_fam_samples(s2, fam, samples_rel)
```

Comment:
I struggled with this last time, but I convinced myself that you don't need a return since you're modifying in place, no? But I'm also terrible at recursion, so you're probably right.